Sources/OpenCombineFoundation/Publishers+Timer.swift

```diff
@@ -0,0 +1,345 @@
+      public let mode: RunLoop.Mode
+      public let options: RunLoop.SchedulerOptions?
+
+      private lazy var routingSubscription: RoutingSubscription = {
+        RoutingSubscription(parent: self)
+      }()
+
+      // Stores if a `.connect()` happened before subscription, 
+      // internally readable for tests
+      internal var isConnected: Bool {
+        return routingSubscription.isConnected
+      }
+
+      /// Creates a publisher that repeatedly emits the current date 
+      /// on the given interval.
+      ///
+      /// - Parameters:
+      ///   - interval: The interval on which to publish events.
+      ///   - tolerance: The allowed timing variance when emitting events. 
+      ///     Defaults to `nil`, which allows any variance.
+      ///   - runLoop: The run loop on which the timer runs.
+      ///   - mode: The run loop mode in which to run the timer.
+      ///   - options: Scheduler options passed to the timer. Defaults to `nil`.
+      public init(
+        interval: TimeInterval,
+        tolerance: TimeInterval? = nil,
+        runLoop: RunLoop, mode: RunLoop.Mode, options: RunLoop.SchedulerOptions? = nil
+      ) {
+        self.interval = interval
+        self.tolerance = tolerance
+        self.runLoop = runLoop
+        self.mode = mode
+        self.options = options
+      }
+
+      /// Adapter subscription to allow `Timer` to multiplex to multiple subscribers
+      /// the values produced by a single `TimerPublisher.Inner`
+      private class RoutingSubscription: Subscription, Subscriber,
+      CustomStringConvertible, CustomReflectable, CustomPlaygroundDisplayConvertible {
+        typealias Input = Date
+        typealias Failure = Never
+
+        private typealias ErasedSubscriber = AnySubscriber<Output, Failure>
+
+        private let lock: Lock
+
+        // Inner is IUP due to init requirements
+        // swiftlint:disable:next implicitly_unwrapped_optional
+        private var inner: Inner<RoutingSubscription>!
+        private var subscribers: [ErasedSubscriber] = []
+
+        private var _lockedIsConnected = false
+        var isConnected: Bool {
+          get {
+            lock.lock()
+            defer { lock.unlock() }
+            return _lockedIsConnected
+          }
+
+          set {
+            lock.lock()
+            let oldValue = _lockedIsConnected
+            _lockedIsConnected = newValue
+
+            // Inner will always be non-nil
+            let inner = self.inner!
+            lock.unlock()
+
+            guard newValue, !oldValue else {
+              return
+            }
+            inner.enqueue()
+          }
+        }
+
+        var description: String { return "Timer" }
+        var customMirror: Mirror { return inner.customMirror }
+        var playgroundDescription: Any { return description }
+        var combineIdentifier: CombineIdentifier { return inner.combineIdentifier }
+
+        init(parent: TimerPublisher) {
+          lock = Lock.allocate()
```

Comment:
Let's move this initialization to the property declaration like we do it in other implementations.